### PR TITLE
Simple parameter expansion + quote removal + field splitting

### PR DESCRIPTION
**Pull request by lsimonne**
```
[ChezLouise@ ~/Documents/42/42sh]$ echo $USER $USER
  executing builtin echo
ChezLouise ChezLouise
  done executing builtin echo, ok
[ChezLouise@ ~/Documents/42/42sh]$ echo "bonjour $USER"
  executing builtin echo
bonjour ChezLouise
  done executing builtin echo, ok
[ChezLouise@ ~/Documents/42/42sh]$ echo '$USER'
  executing builtin echo
$USER
  done executing builtin echo, ok
[ChezLouise@ ~/Documents/42/42sh]$ echo \$USER
  executing builtin echo
$USER
  done executing builtin echo, ok
[ChezLouise@ ~/Documents/42/42sh]$ echo $foo
  executing builtin echo

  done executing builtin echo, ok
```

### CODE REVIEW

File: srcs/expansion/quote_removal.c [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+static void	remove_backslash(t_string **l_addr, t_string **l_str, \
+								bool *quoted, size_t *pos)
+{
+
+	if (!quoted[0] && (quoted[1] && (*l_str)->next && ((*l_str)->next->c == '\"' \
+		|| (*l_str)->next->c == '\\')))
+	{
+		*l_str = (*l_str)->next;
+		list_pop_at_pos(*pos, (t_abstract_list **)l_addr);
```
Comment (pilespin) [inline srcs/expansion/quote_removal.c]: list_pop_at_pos il return le pointeur pour le free

**Comment by pilespin** [comment]
[segv@minus42sh]$ echo "\""
zsh: segmentation fault  ./42sh

**Comment by n0izn0iz** [comment]
j'ai pas compris ton msg @pilespin

File: includes/strlist.h [hunk 2]
```diff
@@ -1,10 +1,13 @@
 #ifndef STRLIST_H
 # define STRLIST_H
 
+#include <stdbool.h>
+
 typedef struct		s_strlist
 {
 	struct s_strlist	*next;
 	char				*str;
+	bool				to_split;
```
Comment (n0izn0iz) [inline includes/strlist.h]: remove to_split?

File: srcs/break_input/substitution.c [hunk 3]
```diff
@@ -5,7 +5,8 @@
 
 bool		is_substitution_start(char const *s)
 {
-	if (s && *s == '`')
+	// does the tokenizer take into account the quoting for start? (it does for end - below)
```
Comment (n0izn0iz) [inline srcs/break_input/substitution.c]: remove comment?

**Comment by pilespin** [comment]
echo "\\"" segfault random

**Comment by pilespin** [comment]
```
echo `&`
42sh: &: No such file or directory
42sh: fatal error: error in add_substitution() in split_subsitutions()
```
exit shell

**Comment by lsimonne** [comment]
command substitution problem, nothing to do with parameter expansion or quote removal

**Pull request merged**